runtime-modules/content/src/lib.rs

```diff
@@ -970,11 +1045,12 @@ decl_module! {
 
             // upload to storage
             if let Some(upload_assets) = params.assets.as_ref() {
-                Self::upload_assets_to_storage(
+                let params = Self::construct_upload_parameters(
                     upload_assets,
                     &channel_id,
-                    &sender,
-                )?;
+                    &sender
+                );
+                 Storage::<T>::upload_data_objects(params)?;
```

Comment:
Missing `can_upload_data_objects` before mutation-safe?

Reply:
Checks for `upload_data_objects` are also inside the `can_create_bag_with_objects`. This is required also because `can_upload_data_objects` throws if the bag does not exist

Reply:
added a comment in 5dc18dc1a4

Reply:
The comment was about `create_video`, not `create_channel`

Reply:
In the `create_video` extrinsic the `upload_data_objects` verifies parameters before uploading. so It will not leave content/storage state in a non valid status.
I will add a comment